src/command.coffee

```diff
@@ -0,0 +1,165 @@
+      @custom.cwd = process.cwd()
+
+    if not @custom.argv or not Array.isArray @custom.argv
+      @custom.argv = process.argv.slice(2)
+
+    if not (@ instanceof DreddCommand)
+      return new DreddCommand(options, @cb)
+
+    @finished = false
+
+  warmUp: ->
+    @finished = false
+
+    @argv = optimist(@custom['argv'], @custom['cwd']).usage(
+      """
+      Usage:
+        dredd <path or URL to blueprint> <api_endpoint> [OPTIONS]
+
+      Example:
+        dredd ./apiary.md http://localhost:3000 --dry-run
+      """
+      )
+      .options(Dredd.options)
+      .wrap(80)
+      .argv
+
+    argError = false
+
+    if not @cb
+      @sigIntEventAdd = true
+      @_processExit = @exit or process.exit
+    else
+      @_processExit = ((code) ->
+        if @finished then return
+        @finished = true
+        if @sigIntEventAdded
+          process.removeEventListener 'SIGINT', @commandSigInt
+        return @cb code
+      ).bind @
+
+    if @argv.help is true
+      optimist.showHelp(console.error)
+      return @_processExit(0)
+
+    if @argv.version is true
+      console.log version
+      return @_processExit(0)
+
+    if not @argv._[0]?
+      console.error("\nError: Must specify path to blueprint file.")
+      argError = true
+
+    if not @argv._[1]?
+      console.error("\nError: Must specify api endpoint.")
+      argError = true
+
+    if argError
+      console.error("\n")
+      optimist.showHelp(console.error)
+      return @_processExit(1)
+
+    # transform path and p argument to array if it's not
+    if !Array.isArray(@argv['path'])
+      @argv['path'] = @argv['p'] = [@argv['path']]
+
+    [configurationForDredd, customConfig] = @initConfig()
+    @dreddInstance = @initDredd configurationForDredd, customConfig
+    return @
+
+  lastArgvIsApiEndpoint: ->
+    # some shells are automatically expanding globs and concating result as arguments
+    # so I'm taking last argument as API endpoint server URL and removing it forom optimist's args
+    @server = @argv._[@argv._.length - 1]
+    @argv._.splice(@argv._.length - 1, 1)
+    return @
+
+  takeRestOfParamsAsPath: ->
+    # and rest of arguments concating to 'path' and 'p' opts, duplicates are filtered out later
+    @argv['p'] = @argv['path'] = @argv['path'].concat(@argv._)
+    return @
+
+  initConfig: ->
+    @
+    .lastArgvIsApiEndpoint()
+    .takeRestOfParamsAsPath()
+
+    configuration =
+      'server': @server
+      'options': @argv
+
+    # push first argument (without some known configuration --key) into paths
+    configuration.options.path ?= []
+    configuration.options.path.push @argv._[0]
+
+    configuration.options.custom ?= {}
+    configuration.options.custom['env'] = @custom['env'] or process.env
+    configuration.options.custom['stdout'] = @custom['stdout'] or process.stdout
+    configuration.options.custom['cwd'] = @custom['cwd'] or process.cwd()
+
+    custom = configuration.options.custom
```

Comment:
This does not make sense for me. Why it can't custom config passed in the configuration object? Isn't it extension of the interface only for purpose of testing?